python/src/scipp/compat/mantid.py

```diff
@@ -328,11 +329,19 @@ def load(filename="",
         mantid.LoadInstrument(ws,
                               FileName=instrument_filename,
                               RewriteSpectraMap=True)
+
+    dataset = None
     if ws.id() == 'Workspace2D':
-        return convert_Workspace2D_to_dataset(ws)
-    if ws.id() == 'EventWorkspace':
-        return convert_EventWorkspace_to_dataset(ws, load_pulse_times,
-                                                 EventType)
-    if ws.id() == 'TableWorkspace':
-        return convert_TableWorkspace_to_dataset(ws, error_connection)
-    raise RuntimeError('Unsupported workspace type')
+        dataset = convert_Workspace2D_to_dataset(ws)
+    elif ws.id() == 'EventWorkspace':
+        dataset = convert_EventWorkspace_to_dataset(ws, load_pulse_times,
+                                                    EventType)
+    elif ws.id() == 'TableWorkspace':
+        dataset = convert_TableWorkspace_to_dataset(ws, error_connection)
+
+    AnalysisDataService.Instance().remove(ws.name())
```

Comment:
Are there and relevant cases where `Load` actually creates multiple workspaces, e.g., when monitors are loaded?

Reply:
Yes, however the converter would break in these situations as it is (the return value of `Load` would no longer be a workspace).

Reply:
:+1:

Reply:
I opened https://github.com/scipp/scipp/issues/677 to record that this is an issue.